app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts

```diff
@@ -0,0 +1,1437 @@
+        this.channelID = channelID;
+        this.rootPostID = rootPostID;
+        this.teamID = teamID;
+        this.intl = intl;
+        this.theme = theme;
+    }
+
+    // composeCallFromCommand creates the form submission call
+    public composeCallFromCommand = async (command: string): Promise<{call: AppCallRequest | null; errorMessage?: string}> => {
+        let parsed = new ParsedCommand(command, this, this.intl);
+
+        const commandBindings = this.getCommandBindings();
+        if (!commandBindings) {
+            return {call: null,
+                errorMessage: this.intl.formatMessage({
+                    id: 'apps.error.parser.no_bindings',
+                    defaultMessage: 'No command bindings.',
+                })};
+        }
+
+        parsed = await parsed.matchBinding(commandBindings, false);
+        parsed = parsed.parseForm(false);
+        if (parsed.state === ParseState.Error) {
+            return {call: null, errorMessage: parserErrorMessage(this.intl, parsed.error, parsed.command, parsed.i)};
+        }
+
+        await this.addDefaultAndReadOnlyValues(parsed);
+
+        const missing = this.getMissingFields(parsed);
+        if (missing.length > 0) {
+            const missingStr = missing.map((f) => f.label).join(', ');
+            return {call: null,
+                errorMessage: this.intl.formatMessage({
+                    id: 'apps.error.command.field_missing',
+                    defaultMessage: 'Required fields missing: `{fieldName}`.',
+                }, {
+                    fieldName: missingStr,
+                })};
+        }
+
+        return this.composeCallFromParsed(parsed);
+    };
+
+    private async addDefaultAndReadOnlyValues(parsed: ParsedCommand) {
+        if (!parsed.form?.fields) {
+            return;
+        }
+
+        await Promise.all(parsed.form.fields.map(async (f) => {
+            if (!f.value) {
+                return;
+            }
+
+            if (f.readonly || !(f.name in parsed.values)) {
+                switch (f.type) {
+                    case AppFieldTypes.TEXT:
+                        parsed.values[f.name] = f.value as string;
+                        break;
+                    case AppFieldTypes.BOOL:
+                        parsed.values[f.name] = 'true';
+                        break;
+                    case AppFieldTypes.USER: {
+                        const userID = (f.value as AppSelectOption).value;
+                        let user: UserModel | UserProfile = (await queryUsersById(this.database, [userID]))[0];
+                        if (!user) {
+                            const res = await fetchUsersByIds(this.serverUrl, [userID]);
+                            if ('error' in res) {
+                            // Silently fail on default value
+                                break;
+                            }
+                            user = res.users[0] || res.existingUsers[0];
+                        }
```

Comment:
This pattern repeats several places in this file (also for channels). The idea is "if the user is not in local, get it from server". The question is... do we want to store the user/channel in local (fetchOnly = [true | false]?)? Reasons not to do it? If we don't belong to that channel, we won't receive updates for it, so we will have stale data.

Reply:
For example:
- I use a command with channel autocomplete, and run the command with ChannelA
- I don't have ChannelA on my database because I am not a member, so I fetch it from the server and store it.
- Someone changes the display name of ChannelA
- I don't receive any websocket event because I am no longer part of the channel.
- If I look for any reason into the information in the database, the channel information will be old (the old display name). Also, not sure if there are any methods to remove it from the database.

I think that right now it is not a big deal in terms of old data, since whenever we look for channels where we do not belong, we ask the server, we do not look the database, but still, may be a problem.